.github/workflows/build-frontend.yml

```diff
@@ -18,6 +18,8 @@ jobs:
           node-version: 18.x
       - run: npm ci
         working-directory: ./frontend
+      - run: npm run lint
```

Comment:
I think even if it's a bit less like what's "normal" for fully JS projects that it's better here to run eslint via `pre-commit`, at least in CI -- speciifically via adding [the eslint hook](https://github.com/pre-commit/mirrors-eslint) to the pre-commit-config.yaml.

Otherwise though looks good to me.

Reply:
Alright, now it does linting in pre-commit :)

Reply:
Hooray! LGTM (merge at will) -- the failures I think are dumb rate limiting to GitHub's own runners, so it may require kicking the build again or something, but yeah changewise, lgtm.